### PR TITLE
Test project compilation fixes for Eclipse

### PR DESCRIPTION
**Pull request by iloveeclipse**
This fixes issues #1316 and #1315.
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: test-harness/build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ apply from: "$rootDir/gradle/maven.gradle"
 
 dependencies {
   compileOnly 'junit:junit:4.13'
-  compileOnly project(':spotbugs')
+  testImplementation project(':spotbugs')
```
Comment (KengoTODA) [inline test-harness/build.gradle:7]: Could you consider to add explanations in `CHANGELOG` for this change?  
I think users of `test-harness` and `test-harness-jupiter` needs to this change (from compile-only scope to compile scope).

**Review by KengoTODA — CHANGES_REQUESTED** [review]

File: test-harness/build.gradle (hunk 1 above)
Comment (iloveeclipse) [reply test-harness/build.gradle:7]: I'm not sure how the changed compilation setting here can affect users. Dependencies itself aren't changed. That should be fully transparent for clients.

Comment (iloveeclipse) [reply test-harness/build.gradle:7]: @KengoTODA : anything else is missing?

Comment (iloveeclipse) [reply test-harness/build.gradle:7]: Ping...

Comment (trancexpress) [reply test-harness/build.gradle:7]: Hi @KengoTODA ,

to work on https://github.com/spotbugs/spotbugs/issues/1254, I had to cherry-pick the changes here. Can the changes get merged so I can work with actual ```master```?

Best regards and thanks,
Simeon

Comment (wreulicke) [reply test-harness/build.gradle:7]: @iloveeclipse 
I think changes in `test-harness` affect its dependency.
That dependency will be changed.
`api` configuration makes `test-harness` dependent on that.

Comment (iloveeclipse) [reply test-harness/build.gradle:7]: @wreulicke : not sure what do you mean by that. How this change can affect *clients* of test-harness? I'm not the gradle user, I have no idea why there are multiple dependencies chains invented by gradle, and how that, once a module is compiled, can affect its clients - I really can't understand that. And btw, let assume test-harness clients would be now also depend on spotbugs - what's wrong by that? How that would affect them, they had spotbugs on classpath already anyway?

My problem is that current state on master is not usable with Eclipse, so contributing is a pain, nothing compiles. That is way more severe issue as a theoretical possibility to somehlw break one or two test-harness clients that may exists in the world.

**Comment by KengoTODA** [comment]
Sorry for my late reaction, recently it's a long national holiday in China.

I'll submit another PR to make `CHANGELOG` better.

**Pull request merged**